src/ppl/nn/engines/x86/impls/src/ppl/kernel/x86/fp32/averagepool2d/normal/averagepool2d_nchw_normal_fp32_sse.cpp

```diff
@@ -0,0 +1,446 @@
+            l_dst[1 * dst_c_stride + 1] = l_dst_trans[1 * C_BLK() + 1];
+            l_dst[1 * dst_c_stride + 2] = l_dst_trans[2 * C_BLK() + 1];
+            l_dst[1 * dst_c_stride + 3] = l_dst_trans[3 * C_BLK() + 1];
+            l_dst_trans += unroll_w * C_BLK();
+            l_dst += unroll_w;
+        }
+        if (w & 2) {
+            l_dst[0 * dst_c_stride + 0] = l_dst_trans[0 * C_BLK() + 0];
+            l_dst[0 * dst_c_stride + 1] = l_dst_trans[1 * C_BLK() + 0];
+            l_dst[1 * dst_c_stride + 0] = l_dst_trans[0 * C_BLK() + 1];
+            l_dst[1 * dst_c_stride + 1] = l_dst_trans[1 * C_BLK() + 1];
+            l_dst_trans += 2 * C_BLK();
+            l_dst += 2;
+        }
+        if (w & 1) {
+            l_dst[0 * dst_c_stride + 0] = l_dst_trans[0 * C_BLK() + 0];
+            l_dst[1 * dst_c_stride + 0] = l_dst_trans[0 * C_BLK() + 1];
+        }
+    } else {
+        while (w >= unroll_w) {
+            w -= unroll_w;
+            l_dst[0 * dst_c_stride + 0] = l_dst_trans[0 * C_BLK() + 0];
+            l_dst[0 * dst_c_stride + 1] = l_dst_trans[1 * C_BLK() + 0];
+            l_dst[0 * dst_c_stride + 2] = l_dst_trans[2 * C_BLK() + 0];
+            l_dst[0 * dst_c_stride + 3] = l_dst_trans[3 * C_BLK() + 0];
+            l_dst_trans += unroll_w * C_BLK();
+            l_dst += unroll_w;
+        }
+        if (w & 2) {
+            l_dst[0 * dst_c_stride + 0] = l_dst_trans[0 * C_BLK() + 0];
+            l_dst[0 * dst_c_stride + 1] = l_dst_trans[1 * C_BLK() + 0];
+            l_dst_trans += 2 * C_BLK();
+            l_dst += 2;
+        }
+        if (w & 1) {
+            l_dst[0 * dst_c_stride + 0] = l_dst_trans[0 * C_BLK() + 0];
+        }
+    }
+}
+
+template<ppl::nn::common::PoolingParam::pooling_mode_t pooling_mode, bool ceil_mode>
+static ppl::common::RetCode averagepool2d_n4cx_fp32_sse_impl(
+    const ppl::nn::TensorShape *src_shape,
+    const ppl::nn::TensorShape *dst_shape,
+    const float *src,
+    const int32_t kernel_h,
+    const int32_t kernel_w,
+    const int32_t stride_h,
+    const int32_t stride_w,
+    const int32_t pad_h,
+    const int32_t pad_w,
+    void *temp_buffer,
+    float *dst)
+{
+    const int32_t batch    = src_shape->GetDim(0);
+    const int32_t channels = src_shape->GetDim(1);
+    const int32_t src_h    = src_shape->GetDim(2);
+    const int32_t src_w    = src_shape->GetDim(3);
+    const int32_t dst_h    = dst_shape->GetDim(2);
+    const int32_t dst_w    = dst_shape->GetDim(3);
+
+    const int32_t width = C_BLK();
+    const int32_t src_hw = src_h * src_w;
+    const int32_t dst_hw = dst_h * dst_w;
+    const int32_t src_trans_len = src_h * (src_w + 2 * pad_w) * width;
+    const int32_t dst_len = dst_w * width;
+    const int32_t thread_buf = src_trans_len + dst_len;
+    int32_t pad_c = round_up(channels, width);
+ 
+    __m128 vsum   = _mm_set1_ps(0.0f);
+    int32_t c_eff;
+    float *base_dst;
+    int32_t ch = channels; 
```

Comment:
`ch` out of parallel region may casue conflict among threads.
And it is unnecessary by the way